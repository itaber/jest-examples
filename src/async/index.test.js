const { resolveAsync, rejectAsync } = require('.');

describe('testing async functions', () => {
  test('can test against resolved promise', async () => {
    await expect(resolveAsync()).resolves.toMatchSnapshot();
  });

  test('can test against rejected promise', async () => {
    await expect(rejectAsync()).rejects.toThrowErrorMatchingSnapshot();
  });
});
