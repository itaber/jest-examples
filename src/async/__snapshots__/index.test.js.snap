// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing async functions can test against rejected promise 1`] = `"something went wrong"`;

exports[`testing async functions can test against resolved promise 1`] = `"success"`;
